events/src/test/java/com/app/events/controller/MediaControllerTest.java

```diff
@@ -0,0 +1,110 @@
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.app.events.constants.EventConstants;
+import com.app.events.constants.MediaConstants;
+import com.app.events.constants.UserConstans;
+import com.app.events.dto.LoginDTO;
+import com.app.events.dto.MediaDTO;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment=WebEnvironment.RANDOM_PORT)
+@TestPropertySource("classpath:application-test.properties")
+public class MediaControllerTest {
+
+    private String authTokenAdmin = "";
+    
+	@Autowired
+    private TestRestTemplate restTemplate;
+    
+    @Before
+    public void login() throws Exception{
+        LoginDTO loginDto = new LoginDTO(UserConstans.DB_ADMIN_USERNAME, UserConstans.DB_ADMIN_PASSWORD);
+        ResponseEntity<String> response = restTemplate.postForEntity("/api/login", loginDto, String.class);
+        authTokenAdmin = response.getBody();
+    }
+
+    @Test
+    public void testGetOneMedia() throws Exception {
+    	URI uri = new URI(MediaConstants.URL_PREFIX + "/" + MediaConstants.PERSISTED_MEDIA_ID);
+    	
+        ResponseEntity<MediaDTO> response = restTemplate.getForEntity(uri, MediaDTO.class);
+        MediaDTO found = response.getBody();
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+	    assertNotNull(found);
+	    assertEquals(MediaConstants.PERSISTED_MEDIA_ID, found.getId());
+	    assertEquals(MediaConstants.PERSISTED_MEDIA_PATH, found.getPath());
+    }
+    
+    @Test
+    public void testGetOneMediaThrow() throws Exception {
+    	URI uri = new URI(MediaConstants.URL_PREFIX + "/" + MediaConstants.INVALID_MEDIA_ID);
+    	
+        ResponseEntity<Object> response = restTemplate.getForEntity(uri, Object.class);
+        
+        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());        
+    }
+    
+    @Test
+    public void testGetEventMedias() throws Exception {
+    	URI uri = new URI(MediaConstants.URL_PREFIX + "/event/" + EventConstants.PERSISTED_EVENT_ID);
+    	
+        ResponseEntity<MediaDTO[]> response = restTemplate.getForEntity(uri, MediaDTO[].class);
+        MediaDTO[] found = response.getBody();
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+	    assertNotNull(found);
+	    assertEquals(2, found.length);
+    }
+    
+    @Test
+    public void testCreateMedia() throws Exception {
+    	URI uri = new URI(MediaConstants.URL_PREFIX);
```

Comment:
kako vratis na predhodno stanje bazu nakon izvrsavanja ovog testa?

Mozes da koristis @Transactional + @Rollback(true) ili da injektujes repo pa da obrises to sto si dodala